src/main/java/Duke.java

```diff
@@ -1,10 +1,69 @@
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        ArrayList<Task> list = new ArrayList<>();
+        System.out.println("Hi! I'm\n" + logo);
+        String effect = "************************************************************";
+        System.out.println(effect + "\nHi! I'm\n" + logo);
+        System.out.println("What can I do for you?\n" + effect);
+        Scanner sc = new Scanner(System.in);
+        String command = sc.nextLine();
+        System.out.println(command.split(" ")[0]);
+        while (!command.equals("bye")) {
+            CommandChecker cc = new CommandChecker();
+            try {
+                cc.checkCommand(command, list);
+                listAction(list, command, effect);
+            } catch (DukeException e) {
+                System.err.println(e);
+            }
+            command = sc.nextLine();
+        }
+        System.out.println(effect + "\n" + "Bye! :) Hope to see you again soon!\n" + effect);
+    }
+
+    public static ArrayList<Task> listAction(ArrayList<Task> list, String command, String effect) {
+        System.out.println(effect);
+        if (command.equals("list")) {
+            System.out.println("Here are the tasks in your list:");
+            int index = 1;
+            for (Task t : list) {
+                System.out.println(index + "." + t);
+                index++;
+            }
+        } else if (command.split(" ")[0].equals("done")) {
+            int target = Integer.parseInt(command.substring(5));
+            Task taskDone = list.get(target - 1);
+            taskDone.markDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(taskDone);
+        } else if (command.split(" ")[0].equals("delete")) {
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(list.get(Integer.parseInt(command.substring(7)) - 1));
+            list.remove(Integer.parseInt(command.substring(7)) - 1);
+            System.out.println("Now you have " + list.size() + " tasks in the list.");
+        }
+        else {
+            if (command.startsWith("deadline")) {
+                list.add(new Deadline(command.substring(9).split("/by")[0], command.substring(9).split("/by")[1]));
+            } else if (command.startsWith("todo")) {
+                list.add(new Todo(command.substring(5)));
+            } else if (command.startsWith("event")) {
+                list.add(new Event(command.substring(6).split("/at")[0], command.substring(6).split("/at")[1]));
+            }
+            System.out.println("Got it! I've added this task:\n" + list.get(list.size() - 1));
+            System.out.println("Now you have " + list.size() + " tasks in the list.");
+        }
+        System.out.println(effect);
+        return new ArrayList<>(list);
     }
 }
+
+
```

Comment:
generally this file looks well written! I can understand it easily. Just a small suggestion, you may want to create a Parser class that will handle all your manipulation of the strings for you:)